pkgs/applications/video/mpv/scripts/default.nix

```
@@ -1,11 +1,63 @@
 { lib
 , callPackage
 , config
+, runCommand
 }:
 
-let buildLua = callPackage ./buildLua.nix { };
-in lib.recurseIntoAttrs
-  ({
+let
+  buildLua = callPackage ./buildLua.nix { };
+
+  inherit (lib.attrsets) filterAttrs optionalAttrs recursiveUpdate unionOfDisjoint;
+  unionOfDisjoints = lib.fold unionOfDisjoint {};
+
+  addTests = name: drv: let
+    inherit (drv) scriptName;
+    scriptPath = "share/mpv/scripts/${scriptName}";
+    fullScriptPath = "${drv}/${scriptPath}";
+
+  in recursiveUpdate drv { passthru.tests = unionOfDisjoints [
+    (drv.passthru.tests or {})
+
+    {
+      scriptName-is-valid = runCommand "mpvScripts.${name}.passthru.tests.scriptName-is-valid" {
+        meta.maintainers = with lib.maintainers; [ nicoo ];
+        preferLocalBuild = true;
+      } ''
+        if [ -f "${fullScriptPath}" ] || [ -d "${fullScriptPath}" ]; then
```

```suggestion
        if [ -e "${fullScriptPath}" ]; then
```

This will also support symlinks